### PR TITLE
jit: Optimize call_light_bif

### PR DESCRIPTION
**Pull request by jhogberg**

### CODE REVIEW

**Review by garazdawi — APPROVED** [review]
Intel changes look good to me

**Comment by max-au** [comment]
May I ask to test this with "extra" microstate accounting states? We rely on the feature and existing CI does not run such configuration.

**Review by bjorng — APPROVED** [review]
Looks good to me.

File: erts/emulator/beam/erl_msacc.h [hunk 1]
```diff
@@ -431,9 +432,10 @@ void erts_msacc_set_bif_state(ErtsMsAcc *msacc, Eterm mod, void *addr);
 #define ERTS_MSACC_SET_STATE_CACHED_M_X(state) ERTS_MSACC_SET_STATE_CACHED_M(state)
 #define ERTS_MSACC_POP_STATE_M_X() ERTS_MSACC_POP_STATE_M()
 #define ERTS_MSACC_PUSH_AND_SET_STATE_M_X(state) ERTS_MSACC_PUSH_AND_SET_STATE_M(state)
-#define ERTS_MSACC_SET_BIF_STATE_CACHED_X(Mod,Addr)       \
-    if (ERTS_MSACC_IS_ENABLED_CACHED_X())               \
-        erts_msacc_set_bif_state(__erts_msacc_cache, Mod, Addr)
+#define ERTS_MSACC_SET_BIF_STATE_CACHED_X(Mod,Addr)                       \
+    if (ERTS_MSACC_IS_ENABLED_CACHED_X()) {                               \
+        (void)erts_msacc_set_bif_state(__erts_msacc_cache, Mod, Addr)     \
+    }
```
Comment (garazdawi) [inline erts/emulator/beam/erl_msacc.h:438]:
```suggestion
#define ERTS_MSACC_SET_BIF_STATE_CACHED_X(Mod,Addr) do {                  \
    if (ERTS_MSACC_IS_ENABLED_CACHED_X()) {                               \
        (void)erts_msacc_set_bif_state(__erts_msacc_cache, Mod, Addr);    \
    } \
    } while(0)
```

Nightly tests are failing with msacc extra. I think the above fix is correct.

**Comment by garazdawi** [comment]
> May I ask to test this with "extra" microstate accounting states? We rely on the feature and existing CI does not run such configuration.

Since the last time msacc extra broke, I've added it to our nightly tests.

File: erts/emulator/beam/erl_msacc.h (hunk 1 above)
Comment (jhogberg) [reply erts/emulator/beam/erl_msacc.h]: Woops! :D

Thanks!

**Pull request merged**